README.md

```diff
@@ -97,53 +97,68 @@ You can also use it within the browser; install via npm and use the `check-error
-var caughtErr;
-
-try {
-  funcThatThrows();
-} catch(e) {
-  caughtErr = e;
-}
-
-checkError.compatibleConstructor(caughtErr, Error); // true
-checkError.compatibleConstructor(caughtErr, TypeError); // true
-checkError.compatibleConstructor(caughtErr, RangeError); // false
+var criteria;
+
+criteria = checkError.createCriteria();
+checkError.describeExpectedError(criteria); // "an Error"
+checkError.checkError(new Error('i like waffles'), criteria); // true
+checkError.checkError(new TypeError('i like waffles'), criteria); // true
+checkError.checkError({ message: 'i like waffles' }, criteria); // false
+
+criteria = checkError.createCriteria('waffles');
+checkError.describeExpectedError(criteria); // "an Error including 'waffles'"
+checkError.checkError(new Error('i like waffles'), criteria); // true
+checkError.checkError(new Error('i like pancakes'), criteria); // false
+checkError.checkError(new TypeError('i like waffles'), criteria); // true
+checkError.checkError({ message: 'i like waffles' }, criteria); // false
+
+criteria = checkError.createCriteria(/waffles/);
+checkError.describeExpectedError(criteria); // "an Error matching /waffles/"
+checkError.checkError(new Error('i like waffles'), criteria); // true
+checkError.checkError(new Error('i like pancakes'), criteria); // false
+checkError.checkError(new TypeError('i like waffles'), criteria); // true
+checkError.checkError({ message: 'i like waffles' }, criteria); // false
+
+criteria = checkError.createCriteria(TypeError);
+checkError.describeExpectedError(criteria); // "a TypeError"
+checkError.checkError(new Error('i like waffles'), criteria); // false
+checkError.checkError(new TypeError('i like waffles'), criteria); // true
+checkError.checkError({ message: 'i like waffles' }, criteria); // false
+
+criteria = checkError.createCriteria(TypeError, 'waffles');
+checkError.describeExpectedError(criteria); // "a TypeError including 'waffles'"
+checkError.checkError(new Error('i like waffles'), criteria); // false
+checkError.checkError(new TypeError('i like waffles'), criteria); // true
+checkError.checkError(new TypeError('i like pancakes'), criteria); // false
+checkError.checkError({ message: 'i like waffles' }, criteria); // false
+
+criteria = checkError.createCriteria(TypeError, /waffles/);
+checkError.describeExpectedError(criteria); // "a TypeError matching /waffles/"
+checkError.checkError(new Error('i like waffles'), criteria); // false
+checkError.checkError(new TypeError('i like waffles'), criteria); // true
+checkError.checkError(new TypeError('i like pancakes'), criteria); // false
+checkError.checkError({ message: 'i like waffles' }, criteria); // false
+
+var errObj = new Error('I like waffles');
+criteria = checkError.createCriteria(errObj);
+checkError.describeExpectedError(criteria); // "[Error: I like waffles]"
+checkError.checkError(new Error('i like waffles'), criteria); // false
+checkError.checkError(errObj, criteria); // true
```

Comment:
I don't have time to give this the _full_ review, but just from glancing at the readme I'm a _little bit_ concerned over the notion of a `criteria` object. My initial questions are:

 - Why do I, as a consumer of this API, need to create an object before asserting on my error.
 - Why can't I just pass some arguments in place of `criteria`. E.g. `checkError.checkError(errObj, new Error('I like waffles')` or `checkError.checkError(errObj, TypeError, /waffles/)`

Reply:
The `.createCriteria` method performs quite a bit of type validation and argument handling (since two method signatures are supported) that is needed by both `.checkError` and `.describeExpectedError`. So the motivation behind the criteria object is only performing that work once. This may be over-engineered. It may be better overall to just bite the bullet and perform the work twice for the purposes of a simpler API. Easy change but will wait for more feedback before taking action.

Reply:
I figured this was the case; it just feels a little un-ergonomic to have to create arguments using this API, to then give that argument to another part of the API (although not unprecedented, see `DataView`).

Perhaps we could make the functions that take a `criteria` polymorphic enough to either accept a criteria, or a criteria signature? This way we get the benefits of both.

We could do something like:

```js
// Each method that wants a `criteria` will also take a criteria signature and coerce as necessary:
function checkError(errObj, ...criteriaLike) {
  const criteria = createCriteria(...criteriaLike)
  // ...
}

// `createCriteria(<Criteria>)` is essentially `identity`
function createCriteria(errLike, errMsgMatcher) {
  if (arguments.length === 1 && isCriteria(errLike) return errLike
  // ...
}
```

Reply:
Seems reasonable to me.

Reply:
@keithamus Node v4 is still supported until April. Do we need to implement that logic with ES5?

Reply:
Sure, the arity semantics are likely of no concern so some sensible variant of `function checkError(errObj, criteriaLikeA, criteriaLikeB)` will be fine, right?

Reply:
Excellent idea.
I totally agree with @keithamus. This makes the API a lot more straightforward.